api/web3server.go

```diff
@@ -468,56 +469,22 @@ func (svr *Web3Server) sendRawTransaction(in interface{}) (interface{}, error) {
 	if err != nil {
 		return nil, err
 	}
-
-	// load the value of gasPrice, value, to
-	gasPrice, value, to := "0", "0", ""
-	if tx.GasPrice() != nil {
-		gasPrice = tx.GasPrice().String()
-	}
-	if tx.Value() != nil {
-		value = tx.Value().String()
-	}
-	if tx.To() != nil {
-		ioAddr, _ := address.FromBytes(tx.To().Bytes())
-		to = ioAddr.String()
-	}
-
-	req := &iotextypes.Action{
-		Core: &iotextypes.ActionCore{
-			Version:  0,
-			Nonce:    tx.Nonce(),
-			GasLimit: tx.Gas(),
-			GasPrice: gasPrice,
-			ChainID:  svr.coreService.ChainID(),
-		},
-		SenderPubKey: pubkey.Bytes(),
-		Signature:    sig,
-		Encoding:     iotextypes.Encoding_ETHEREUM_RLP,
+	act, err := svr.ethTxToAction(tx)
+	if err != nil {
+		return nil, err
 	}
 
-	// TODO: process special staking action
-
-	if isContract, _ := svr.isContractAddr(to); !isContract {
-		// transfer
-		req.Core.Action = &iotextypes.ActionCore_Transfer{
-			Transfer: &iotextypes.Transfer{
-				Recipient: to,
-				Payload:   tx.Data(),
-				Amount:    value,
-			},
-		}
-	} else {
-		// execution
-		req.Core.Action = &iotextypes.ActionCore_Execution{
-			Execution: &iotextypes.Execution{
-				Contract: to,
-				Data:     tx.Data(),
-				Amount:   value,
-			},
-		}
-	}
+	elp := (&action.EnvelopeBuilder{}).SetVersion(1).SetNonce(tx.Nonce()).SetGasLimit(tx.Gas()).SetGasPrice(tx.GasPrice()).
+		SetChainID(svr.coreService.ChainID()).SetAction(act).Build()
+	selp := action.NewSealedEnvelope(
+		elp,
+		iotextypes.Encoding_ETHEREUM_RLP,
+		config.EVMNetworkID(),
+		pubkey,
+		sig,
+	)
```

Comment:
here should call `NewStakingActionFromABIBinary()` to check web3 staking tx, and wrap into the execution (with special address)?